File: src/main/java/wraith/fabricaeexnihilo/modules/infested/InfestedLeavesItem.java
package wraith.fabricaeexnihilo.modules.infested;

import net.fabricmc.fabric.api.item.v1.FabricItemSettings;
import net.minecraft.item.BlockItem;
import wraith.fabricaeexnihilo.modules.base.Colored;
import wraith.fabricaeexnihilo.util.Color;

public class InfestedLeavesItem extends BlockItem implements Colored {
    
    public InfestedLeavesItem(InfestedLeavesBlock block, FabricItemSettings settings) {
        super(block, settings);
    }
    
    @Override
    public int getColor(int index) {
        return Color.WHITE.toInt();
    }
}

File: src/main/kotlin/exnihilofabrico/common/infested/InfestedLeavesItem.kt
package exnihilofabrico.common.infested

import exnihilofabrico.common.base.IHasColor
import exnihilofabrico.util.Color
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment
import net.minecraft.item.BlockItem
import net.minecraft.item.ItemStack
import net.minecraft.text.Text
import net.minecraft.text.TranslatableText

class InfestedLeavesItem(block: InfestedLeavesBlock, settings: Settings): BlockItem(block, settings), IHasColor {
    override fun getColor(index: Int) = Color.WHITE.toInt()

    @Environment(EnvType.CLIENT)
    override fun getName() = TranslatableText("block.exnihilofabrico.infested", (this.block as InfestedLeavesBlock).leafBlock.name)
    override fun getName(stack: ItemStack): Text = this.name
}